aten/src/ATen/native/Blas.cpp

```diff
@@ -82,23 +90,17 @@ TORCH_IMPL_FUNC(addmv_out_cpu)(const Tensor &self, const Tensor &mat, const Tens
   }
 }
 
-Tensor &mv_out(const Tensor &self, const Tensor &vec, Tensor& result) {
+TORCH_IMPL_FUNC(mv_out)(const Tensor &self, const Tensor &vec,  const Tensor& result) {
   //self arg sent to addmv_out cannot be resized
   //here we use result as self argument for addmv, and result is user supplied and can be wrong size
   //it's not a hard error, because we allow resizing result, but it becomes a hard error
   //in addmv, because addmv expects self to satisfy proper conditions
   //to avoid this, supply correctly sized self, its contents doesn't matter because beta is 0
   if (result.dim() > 1 || (result.numel() != self.size(0) || result.numel() !=1)) {
     Tensor self_addmv = at::empty({self.size(0)}, self.options());
-    return at::addmv_out(result, self_addmv, self, vec, 0, 1);
+    at::addmv_out(const_cast<Tensor&>(result), self_addmv, self, vec, 0, 1);
   }
-  return at::addmv_out(result, result, self, vec, 0, 1);
-}
-
-Tensor mv(const Tensor &self, const Tensor &vec) {
-  Tensor result = at::empty({self.size(0)}, self.options());
-  //inplace version is more efficient if we can use it
-  return at::addmv_(result, self, vec, 0, 1);
```

Comment:
@ngimel i think you added this previously. but i think this PR will make this not useful anymore?

Reply:
set_output of inplace version is slightly more efficient than that of `out` version (inplace just sets self as the output, `out` checks out's device and dtype, so calling inplace when possible still makes sense.